design/cstor/v1/pool_expansion_by_expanding_disk.md

```diff
@@ -0,0 +1,145 @@
+
+- BlockDevice replacement:
+	cstor-pool-mgmt will iterate over all the blockdevices present in CstorPoolInstance and the verifies is that blockdevice is for replacement(We are maintaining annotation called openebs.io/predecessor: <replaced_blockdevice_name> on the claim of that blockdevice) if that block device is for replacement then the it will perform replacement operation on the pool.
+
+### Proposed Implementations
+
+CStor-pool-mgmt has CSPI controller for every sync operation cstor-pool-mgmt will iterate over all the blockdevices exist on CSPI by iterating over CstorPoolInstanceBlockDevices of all raid groups. Now if the capacity exist on blockdevice CR is greater than the  capacity exists on CstorPoolInstanceBlockDevice then we can follow the steps to perform the pool expansion. But anyways it is not good practice to perform expansion of pool with out getting the permission from user. So to get the input from the user we will be introducing a new field under CstorPoolInstanceBlockDevice called IsBlockDeviceExpanded. If  IsBlockDeviceExpanded sets true then it is something like user/admin is requesting to perform pool expansion steps. If IsBlockDeviceExpanded sets false then NoOps will be performed.
+
+Currently CstorPoolInstanceBlockDevice holds below configurations 
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+}
+```
+
+To get user Input proposed to add field in above struct
+
+```go
+// CStorPoolInstanceBlockDevice contains the details of block devices that
+// constitutes a raid group.
+type CStorPoolInstanceBlockDevice struct {
+	// BlockDeviceName is the name of the block device.
+	BlockDeviceName string `json:"blockDeviceName"`
+
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+
+	// IsBlockDeviceExpanded informs whether blockdevice 
+	// got expanded or not
+	IsBlockDeviceExpanded bool `json:”isBlockDeviceExpanded”`
+}
+```
+
+### Steps to perform user stories
+- User has to set the value of IsBlockDeviceExpanded to true in corresponding CStorPoolInstanceBlockDevice.
+
+#### Pool Expansion Steps:
+
+Step1: Set autoexpand to on in pool by triggering `zpool set autoexpand=on <pool_name>`.
+
+Step2: Remove the partition if any exists which was created by cstor-pool container(ex: sdb9).
```

Comment:
Remove the buffering partition if exists which was created by cstor-pool container once the disk get expanded.